cypress/integration/workflow-executions.spec.js

```diff
@@ -1,43 +1,86 @@
 /// <reference types="cypress" />
 
-const waitOnAPIs = () => {
-  cy.wait('@list-workflows');
-};
+const statuses = [
+  'Running',
+  'TimedOut',
+  'Completed',
+  'Failed',
+  'ContinuedAsNew',
+  'Canceled',
+  'Terminated',
+];
 
 describe('Workflow Executions List', () => {
   beforeEach(() => {
-    cy.visit('/namespaces/default/workflows');
-
-    cy.intercept(Cypress.env('VITE_API_HOST') + '/api/v1/namespaces?*', {
-      fixture: 'namespaces.json',
-    }).as('namespaces-api');
+    cy.interceptApi();
 
-    cy.intercept(
-      Cypress.env('VITE_API_HOST') +
-        '/api/v1/namespaces/default/workflows?query=*',
-      { fixture: 'workflows.json' },
-    ).as('list-workflows');
-  });
+    cy.visit('/namespaces/default/workflows');
 
-  it('should call the namespaces when starting up', () => {
-    waitOnAPIs();
+    cy.wait('@workflows-api');
+    cy.wait('@namespaces-api');
   });
 
   it('should default to 24 hours for the time range', () => {
-    waitOnAPIs();
-
-    cy.url().should('contain', '?time-range=24+hours');
+    cy.url().should('contain', 'time-range=24+hours');
 
     cy.get('#time-range-filter')
       .find('option:selected')
       .should('have.value', '24 hours');
   });
 
   it('should default to showing all workflows', () => {
-    waitOnAPIs();
-
     cy.get('#status-filter')
       .find('option:selected')
       .should('have.value', 'null');
+
+    cy.get('#workflow-id-filter').should('have.value', '');
+
+    cy.get('#workflow-type-filter').should('have.value', '');
+  });
+
+  describe('Workflow Filters', () => {
+    it('should send the correct query for Workflow Type', () => {
+      const result = encodeURIComponent('WorkflowType="ImportantWorkflowType"');
+
+      cy.get('#workflow-type-filter').type('ImportantWorkflowType');
+
+      cy.url().should('contain', 'workflow-type=ImportantWorkflowType');
+
+      cy.wait('@workflows-api').then((interception) => {
+        expect(interception.request.url).to.contain(result);
+      });
+    });
+
+    it('should send the correct query for Workflow ID', () => {
+      const result = encodeURIComponent('WorkflowId="002c98_Running"');
+
+      cy.get('#workflow-id-filter').type('002c98_Running');
```

Comment:
how is this linked to an html element? didn't find workflow-id-filter

Reply:
It's created by `FilterSearch`. It takes the query parameter and tacks on `-filter`.